storage/objects/change_file_storage_class.go

```diff
@@ -0,0 +1,59 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package objects
+
+// [START storage_change_file_storage_class]
+import (
+	"context"
+	"fmt"
+	"io"
+	"time"
+
+	"cloud.google.com/go/storage"
+)
+
+// changeObjectStorageClass changes the storage class of objects
+// within a bucket.
+func changeObjectStorageClass(w io.Writer, bucketName, objectName string) error {
+	// bucketName := "bucket-name"
+	// objectName := "object-name"
+	ctx := context.Background()
+	client, err := storage.NewClient(ctx)
+	if err != nil {
+		return fmt.Errorf("storage.NewClient: %v", err)
+	}
+	defer client.Close()
+
+	ctx, cancel := context.WithTimeout(ctx, time.Second*10)
+	defer cancel()
+
+	bucket := client.Bucket(bucketName)
+	object := bucket.Object(objectName)
+	// See the StorageClass documentation for other valid storage classes:
+	// https://cloud.google.com/storage/docs/storage-classes
+	newStorageClass := "COLDLINE"
+	// You can't change an object's storage class directly, the only way is
+	// to rewrite the object with the desired storage class.
+	copier := object.CopierFrom(object)
+	copier.StorageClass = newStorageClass
+	_, err = copier.Run(ctx)
```

Comment:
Shouldn't we delete the old version of object? Rewriting is announced, but in fact it is copying. Or under the hood it replaces the old object with new?

Reply:
I don't understand actually, why old version is important for us in this case. 
We further delete anyway this object. And rewrite is implemented in `copier.Run(ctx)`